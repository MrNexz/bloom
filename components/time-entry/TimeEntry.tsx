import React from 'react';

import './time-entry.scss';
import { TimeEntryModel } from '../../ducks/time-entries';

const isoToLocale = (time) => new Date(time).toLocaleTimeString('nl-NL', { hour: '2-digit', minute: '2-digit' });

interface TimeEntryProps {
  timeEntry: TimeEntryModel;
  onDelete;
}

class TimeEntry extends React.Component<TimeEntryProps> {
  static defaultProps = {
    clientName: 'Client Unknown'
  }

  render () {
    const { clientName, id, stopTime, startTime } = this.props.timeEntry
    return (
      <section className="time-entry">
        <div className="single-time-entry">
          <div className="single-time-entry__wrapper-client-delete">
            <p className="single-time-entry__client">
              {clientName}
            </p>
            <button
              className="single-time-entry__delete"
              type="button"
              onClick={() => this.props.onDelete(id)}
            >
              Delete
              <div className="single-time-entry__forbidden-icon" />
            </button>
          </div>
          <div>
            <p className="single-time-entry__hours">
              {isoToLocale(startTime)} - {isoToLocale(stopTime)}
            </p>
            <p className="single-time-entry__hours-worked">
              {isoToLocale((Date.parse(stopTime)
                - Date.parse(startTime))
                - 1 * 1000 * 60 * 60)}
            </p>
          </div>
        </div>
      </section>
    )
  }
}

export default TimeEntry;
