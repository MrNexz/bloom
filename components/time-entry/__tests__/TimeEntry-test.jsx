import React from 'react';
import { shallow } from 'enzyme';

import TimeEntry from '../TimeEntry.tsx';

const timeEntry = {
  client: 'Humanoids',
  startTime: '2018-08-18T07:00:00.000Z',
  stopTime: '2018-08-18T15:00:00.000Z',
  id: 1
};

const onDelete = () => {};

it('time entry should render without crashing', () => {
  const wrapper = shallow(
    <TimeEntry
      onDelete={onDelete}
      client={timeEntry.client}
      id={timeEntry.id}
      startTime={timeEntry.startTime}
      stopTime={timeEntry.stopTime}
    />
  );
  expect(wrapper).toMatchSnapshot();
});
