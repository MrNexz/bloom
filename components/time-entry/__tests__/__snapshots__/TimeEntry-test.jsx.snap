// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`time entry should render without crashing 1`] = `
<section
  className="time-entry"
>
  <div
    className="single-time-entry"
  >
    <div
      className="single-time-entry__wrapper-client-delete"
    >
      <p
        className="single-time-entry__client"
      >
        Client Unknown
      </p>
      <button
        className="single-time-entry__delete"
        onClick={[Function]}
        type="button"
      >
        Delete
        <div
          className="single-time-entry__forbidden-icon"
        />
      </button>
    </div>
    <div>
      <p
        className="single-time-entry__hours"
      >
        09:00
         - 
        17:00
      </p>
      <p
        className="single-time-entry__hours-worked"
      >
        08:00
      </p>
    </div>
  </div>
</section>
`;
